pkg/volume/csi/csi_plugin.go

```diff
@@ -54,12 +62,27 @@ func ProbeVolumePlugins() []volume.VolumePlugin {
 	return []volume.VolumePlugin{p}
 }
 
+// RegistrationCallback is called by kubelet's plugin watcher upon detection
+// of a new registration socket opened by CSI Driver registrar side car.
+func RegistrationCallback(pluginName string, endpoint string, versions []string, socketPath string) (error, chan bool) {
+
+	glog.Infof("><SB> Callback from kubelet with plugin name: %s endpoint: %s versions: %s socket path: %s",
+		pluginName, endpoint, strings.Join(versions, ","), socketPath)
+
+	// Storing endpoint of newly registered CSI driver into the map, where CSI driver name will be the key
+	// all other CSI components will be able to get the actual socket of CSI drivers by its name.
+	csiDrivers.Store(pluginName, csiDriver{driverEndpoint: endpoint})
```

Comment:
If all type `csiDriver` is storing is a string value, consider just using a `string` directly instead of the extra type.

Reply:
@vladimirvivien yeah, I was thinking more about future extensibility. Adding something to a struct would require less refactoring than converting endpoint string into struct. I do not have strong opinion though.

Reply:
I'm ok with the new type as is. It can can be expanded in the future if needed, and unlike opaque strings can be type checked.